### PR TITLE
Adding gas to System

### PR DESCRIPTION
**Pull request by TRV-PHD**
The purpose of this pull request is to expand the __init__ function to allow gas as a secondary fluid to the system.

### CODE REVIEW

File: flowforge/input/System.py [hunk 1]
```diff
@@ -113,6 +113,7 @@ def __init__(self, components: Dict[str, Component], sysdict: Dict, unitdict: Di
         self._MMBC = None
         self._EBC = None
         self._fluid = None
+        self._gasfluid = None
```
Comment (cgentry7) [inline flowforge/input/System.py:116]: @bscollin, I don't really like having `fluid` and `gasfluid`.  Part of me just thinks `fluid` and `void`.

**Review by cgentry7 — CHANGES_REQUESTED** [review]
Would probably be a good idea to go ahead and add the hooks for void body forces and wall functions.

File: flowforge/input/System.py [hunk 2]
```diff
@@ -162,11 +173,22 @@ def _setupSimpleLoop(
...
+
         """
         self._isLoop = True
 
         components, loop = make_continuous(components, loop)
         self._fluidname = fluid.lower()
+        if gas is not None:
+            self._gasname = gas.lower()
+        else:
+            self._gasname = gas
```
Comment (cgentry7) [inline flowforge/input/System.py:191]: shorter to just say:

`self._gasname = gas if gas is None else gas.lower()`

File: flowforge/input/System.py [hunk 3]
```diff
@@ -206,11 +233,19 @@ def _setupSegment(
...
+            An optional parameter to specify gas in the system (e.g. "Helium")
         """
         self._isLoop = False
 
         components, order = make_continuous(components, order)
         self._fluidname = fluid.lower()
+        if gas is not None:
+            self._gasname = gas.lower()
+        else:
+            self._gasname = gas
```
Comment (cgentry7) [inline flowforge/input/System.py:248]: same as above

File: flowforge/input/BoundaryConditions.py [hunk 4]
```diff
@@ -132,6 +132,42 @@ def _convertUnits(self, uc: UnitConverter) -> None:
...
+    ----------
+    mdot : float
+        Mass float rate [kg/s]
+    """
+
+    def __init__(self, inlet: dict = None, outlet: dict = None):
+        assert inlet or outlet
+
+        self._type_inlet = None
+        self._val_inlet = None
```
Comment (cgentry7) [inline flowforge/input/BoundaryConditions.py:148]: Is it possible to have this as an outlet?  If so, I'd go ahead and add it.

File: flowforge/input/BoundaryConditions.py [hunk 5]
```diff
@@ -132,6 +132,42 @@ def _convertUnits(self, uc: UnitConverter) -> None:
             else:
                 raise Exception("Unknown enthalpy BC type: " + self.type_outlet)
 
+class VoidBC:
+    """Class for void fraction BC
+
+    Attributes
+    ----------
+    mdot : float
+        Mass float rate [kg/s]
```
Comment (cgentry7) [inline flowforge/input/BoundaryConditions.py:141]: Would we want to support `mdot` and `void_fraction`?  Maybe not?  Just wanted to think about it a little.

**Review by cgentry7 — CHANGES_REQUESTED** [review]
Couple of minor things, but overall looks good.

File: flowforge/input/System.py (hunk 2 above)
Comment (TRV-PHD) [reply flowforge/input/System.py:191]: Agreed

File: flowforge/input/System.py (hunk 1 above)
Comment (TRV-PHD) [reply flowforge/input/System.py]: Using fluid and gas

File: flowforge/input/System.py (hunk 3 above)
Comment (TRV-PHD) [reply flowforge/input/System.py:248]: Agreed

File: flowforge/input/BoundaryConditions.py (hunk 4 above)
Comment (TRV-PHD) [reply flowforge/input/BoundaryConditions.py:148]: Yes, added

File: flowforge/input/BoundaryConditions.py (hunk 5 above)
Comment (TRV-PHD) [reply flowforge/input/BoundaryConditions.py:141]: Same as above

File: flowforge/input/BoundaryConditions.py [hunk 6]
```diff
@@ -132,6 +132,64 @@ def _convertUnits(self, uc: UnitConverter) -> None:
...
+        self._type_inlet = None
+        self._val_inlet = None
+        if inlet:
+            if "mdot" in inlet:
+                self._type_inlet = "mdot"
+                self._val_inlet = inlet["mdot"]
+            else:
+                self._type_inlet = "void_fraction"
+                self._val_inlet = inlet["void_fraction"]
+
```
Comment (cgentry7) [inline flowforge/input/BoundaryConditions.py:158]: ok, so as of now, you're not handling viable errors.  You need to add an assertions to make sure a user provides one of the valid types (i.e. `mdot` or `void_fraction`), doesn't provide both, doesn't specify both inlet and outlet, and that `mdot > 0.` or `0. <= void_fraction <= 1.`.

File: flowforge/input/System.py [hunk 7]
```diff
@@ -112,7 +112,11 @@ def __init__(self, components: Dict[str, Component], sysdict: Dict, unitdict: Di
         self._wallfunctions = []
         self._MMBC = None
         self._EBC = None
+        self._VBC = None
         self._fluid = None
+        self._gas = None
+        self._gasbodyforces = []
+        self._gaswallfunctions = []
```
Comment (cgentry7) [inline flowforge/input/System.py:119]: I don't think you need separate body forces / wall functions for gas.  At least not here.  SyTH will split them out when processing `_bodyforces` and `gaswallfunctions`.

Comment (TRV-PHD) [reply flowforge/input/System.py:119]: Agreed

File: flowforge/input/BoundaryConditions.py (hunk 6 above)
Comment (TRV-PHD) [reply flowforge/input/BoundaryConditions.py:158]: image

for MMBC the mdot can be anything but 0

Comment (TRV-PHD) [reply flowforge/input/BoundaryConditions.py:158]: image

Current

File: flowforge/input/System.py (hunk 7 above)
Comment (cgentry7) [reply flowforge/input/System.py]: removed both

**Pull request merged**